### PR TITLE
Updates for Karpenter 0.36.x CRDs

### PR DESCRIPTION
**Pull request by NicholasRaymondiSpot**
# TL:DR;
Updates CRDs-catalog for Karpenter 0.36.x configurations

## Issue
While no changes are needed to get a proper validation with current configurations, there are some changes in 0.36 to consider adding for supported feature changes.

## Solution
I used the [CRD Extractor](https://github.com/InfinityFlowApp/CRDs-catalog?tab=readme-ov-file#crd-extractor) script to export the 0.36 CRDs for Karpenter, there were no EC2NodeClass changes with this release. I based my testing changes on the [Node Pool example provided](https://karpenter.sh/docs/concepts/nodepools/). While the new label doesn't cause a failure, I was able to confirm validation by adding a new kubelet config to our manifest with invalid int values, here's a snippet from the manifest, note the incorrect value types for `cpu` and `memory`:
```
apiVersion: karpenter.sh/v1beta1
kind: NodePool
spec:
  template:
    spec:
      kubelet:
        kubeReserved:
          cpu: 100
          memory: 100
      requirements:
      - key: karpenter.k8s.aws/instance-cpu-manufacturer
        operator: In
        values:
        - "aws"
```

This should fail but doesn't with the currently published CRDs, we get a validated manifest using this command:
```
kubeconform -schema-location default -schema-location 'https://raw.githubusercontent.com/InfinityFlowApp/CRDs-catalog/main/{{.Group}}/{{.ResourceKind}}_{{.ResourceAPIVersion}}.json' "$output_path/$chart_name.yaml"
```

Changes in karpenter-nodepools.yaml:
```
   template:
     metadata: {}
     spec:
+      kubelet:
+        kubeReserved:
+          cpu: 100
+          memory: 100
       nodeClassRef:
         name: testing-node-class
       requirements:
@@ -66,6 +70,10 @@ spec:
         operator: Gt
         values:
         - "2"
+      - key: karpenter.k8s.aws/instance-cpu-manufacturer
+        operator: In
+        values:
+        - "aws"
```

Once I updated my feature branch and ran validation using it, we get the expected failure for using `int` instead of `string`:
```
kubeconform -schema-location default -schema-location 'https://raw.githubusercontent.com/NicholasRaymondiSpot/CRDs-catalog/karpenter-036/{{.Group}}/{{.ResourceKind}}_{{.ResourceAPIVersion}}.json' "$output_path/$chart_name.yaml"
```

Error:
```
NodePool testing-node-pool is invalid: 
  problem validating schema. Check JSON formatting: 
    jsonschema: '/spec/template/spec/kubelet/kubeReserved/memory' does not validate with 
    https://raw.githubusercontent.com/NicholasRaymondiSpot/CRDs-catalog/karpenter-036/karpenter.sh/nodepool_v1beta1.json#/properties/spec/properties/template/properties/spec/properties/kubelet/properties/kubeReserved/additionalProperties/type: expected string, but got number
```

## Changes Made
* NodeClaim & NodePool
** Kublet properties for `kubeReserved` & `systemReserved` are modified to only accept strings
** The requirements spec has an updated validation rule for labels matching `karpenter.k8s.aws/instance-cpu-manufacturer`

## Notes
This was tested with Karpenter 0.36.2

## Resources & References
https://raw.githubusercontent.com/aws/karpenter/v0.35.5/pkg/apis/crds/karpenter.k8s.aws_ec2nodeclasses.yaml
https://raw.githubusercontent.com/aws/karpenter/v0.36.2/pkg/apis/crds/karpenter.k8s.aws_ec2nodeclasses.yaml

https://raw.githubusercontent.com/aws/karpenter/v0.35.5/pkg/apis/crds/karpenter.sh_nodeclaims.yaml
https://raw.githubusercontent.com/aws/karpenter/v0.36.2/pkg/apis/crds/karpenter.sh_nodeclaims.yaml

https://raw.githubusercontent.com/aws/karpenter/v0.35.5/pkg/apis/crds/karpenter.sh_nodepools.yaml
https://raw.githubusercontent.com/aws/karpenter/v0.36.2/pkg/apis/crds/karpenter.sh_nodepools.yaml

[Upgrading to v0.36.0+](https://karpenter.sh/v0.36/upgrading/upgrade-guide/#upgrading-to-0360)

### CODE REVIEW

**Comment by NicholasRaymondiSpot** [comment]
@ElanHasson this is ready for review.
